### PR TITLE
Fix relationships and cardinality

### PR DESCRIPTION
**Pull request by mprahl**
Addresses #62

### CODE REVIEW

**Comment by mprahl** [comment]
@yashvardhannanavati just rebased it.

File: purview/models/distgit.py [hunk 1]
```diff
@@ -26,8 +27,9 @@ class DistGitBranch(PurviewStructuredNode):
     repo_name = StringProperty(required=True)
     repo_namespace = StringProperty(required=True)
     commits = RelationshipTo('DistGitCommit', 'CONTAINS')
+    contributors = RelationshipTo('.user.User', 'CONTRIBUTED_BY')
```
Comment (yashvardhannanavati) [inline purview/models/distgit.py]: Any specific reason as to why this relationship is added? I think this wasn't there before right

File: purview/models/distgit.py [hunk 2]
```diff
@@ -50,15 +51,16 @@ class DistGitCommit(PurviewStructuredNode):
...
-    pushes = RelationshipTo('DistGitPush', 'PUSHED_IN')
-    related_bugs = RelationshipTo('.bugzilla.BugzillaBug', 'RELATED_TO')
-    repos = RelationshipTo('DistGitRepo', 'CONTAINED_BY')
+    author = RelationshipTo('.user.User', 'AUTHORED_BY', cardinality=ZeroOrOne)
+    branches = RelationshipFrom('DistGitBranch', 'CONTAINS')
+    # Cardinality is enforced on the `parent` property, so the `children` property should be
+    # treated as read-only
+    children = RelationshipFrom('.distgit.DistGitCommit', 'PARENT')
+    committer = RelationshipTo('.user.User', 'COMMITTED_BY', cardinality=ZeroOrOne)
+    koji_builds = RelationshipFrom('.koji.KojiBuild', 'BUILT_FROM')
```
Comment (yashvardhannanavati) [inline purview/models/distgit.py]: I was hoping we could keep this as RelationshipTo because of the story flow.

File: purview/models/errata.py [hunk 3]
```diff
@@ -28,15 +29,13 @@ class Advisory(PurviewStructuredNode):
...
-    attached_builds = RelationshipTo('.koji.KojiBuild', 'RELATED_TO')
-    # A normalized relationship name (same as reporters)
-    owners = RelationshipTo('.user.User', 'OWNED_BY')
-    package_owners = RelationshipTo('.user.User', 'PACKAGE_OWNED_BY')
-    reporters = RelationshipTo('.user.User', 'REPORTED_BY')
-    states = RelationshipTo('AdvisoryState', 'RELATED_TO')
-    triggers_freshmaker_event = RelationshipTo('.freshmaker.FreshmakerEvent', 'TRIGGERS')
+    assigned_to = RelationshipTo('.user.User', 'ASSIGNED_TO', cardinality=ZeroOrOne)
+    attached_bugs = RelationshipTo('.bugzilla.BugzillaBug', 'ATTACHED')
+    attached_builds = RelationshipTo('.koji.KojiBuild', 'ATTACHED')
```
Comment (yashvardhannanavati) [inline purview/models/errata.py]: Again, according to the story flow, I was wondering if we could have RelationshipFrom here and RelationshipTo in the KojiBuild model

File: purview/models/koji.py [hunk 4]
```diff
@@ -4,43 +4,45 @@
...
 from purview.models.base import PurviewStructuredNode
 
 
 class KojiBuild(PurviewStructuredNode):
     """Definition of a Koji build in Neo4j."""
 
-    advisories = RelationshipTo('.errata.Advisory', 'RELATED_TO')
-    commits = RelationshipTo('.distgit.DistGitCommit', 'RELATED_TO')
+    advisories = RelationshipFrom('.errata.Advisory', 'ATTACHED')
+    commit = RelationshipTo('.distgit.DistGitCommit', 'BUILT_FROM', cardinality=ZeroOrOne)
```
Comment (yashvardhannanavati) [inline purview/models/koji.py]: If you agree with the comments above, then this would change too

File: purview/models/bugzilla.py [hunk 5]
```diff
@@ -3,32 +3,31 @@
...
-    qa_contacts = RelationshipTo('.user.User', 'QA_BY')
-    related_by_commits = RelationshipTo('.distgit.DistGitCommit', 'RELATED_TO')
-    reporters = RelationshipTo('.user.User', 'REPORTED_BY')
+    qa_contact = RelationshipTo('.user.User', 'QA_BY', cardinality=ZeroOrOne)
+    related_by_commits = RelationshipFrom('.distgit.DistGitCommit', 'RELATED')
+    reporter = RelationshipTo('.user.User', 'REPORTED_BY', cardinality=ZeroOrOne)
     resolution = StringProperty()
-    resolved_by_commits = RelationshipTo('.distgit.DistGitCommit', 'RESOLVED_BY')
-    reverted_by_commits = RelationshipTo('.distgit.DistGitCommit', 'REVERTED_BY')
+    resolved_by_commits = RelationshipFrom('.distgit.DistGitCommit', 'RESOLVED')
```
Comment (yashvardhannanavati) [inline purview/models/bugzilla.py]: Also, this would need to be RelationshipTo here :)

File: purview/models/bugzilla.py [hunk 6]
```diff
@@ -3,32 +3,31 @@
...
+    reverted_by_commits = RelationshipFrom('.distgit.DistGitCommit', 'REVERTED')
```
Comment (yashvardhannanavati) [inline purview/models/bugzilla.py]: same as above

**Comment by mprahl** [comment]
We agreed to keep it as is.

**Comment by yashvardhannanavati** [comment]
:+1:

**Pull request merged**